.github/workflows/variables.yml

```diff
@@ -98,9 +108,13 @@ jobs:
         with:
           result-encoding: string
           script: |
-            if ( "${{ github.event.pull_request.base.ref }}".match(/^(eol|openuchile|vlabx)(-release)?\/(\w+)(\.master)?$/) ) { // pr to image generation branch case
-              return true
-            } else return false
+            if ( "${{ github.ref }}".match(/^refs\/heads\/gh-readonly-queue\//) ) { // merge queue branch case
+              return false;
```

Comment:
La MQ siempre se ejecuta como parte de un PR, por tanto la salida debe ser `true`. Busca para qué se usa y entenderás por qué esta variable es necesaria.

Lo mismo pasa con la variable inmediatamente anterior (`is_release`). Revisar si está bien lo que retorna para las MQ.

Reply:
He profundizado un poco más en el código y he llegado a algunas conclusiones.

is_pr es una parte clave del flujo de trabajo que indica si el evento que lo desencadenó es una solicitud de extracción (PR).

Si el evento es una PR y cumple con cierta condición especificada en la expresión regular dentro del bloque if, se establece como verdadero (true).

¿Qué dice exactamente la condición? Si "github.event.pull_request.base.ref" coincide con el patrón /^(eol|openuchile|vlabx)(-release)?/(\w+)(.master)?$/, entonces se establece como verdadero. Esto significa que la solicitud de PR está dirigida a nuestra rama de producción, que es la que genera la imagen.

Por otro lado, he entendido que is_release es una variable destinada a indicar si la ejecución está ocurriendo en una rama de producción.

Si la rama coincide con el patrón /^refs/heads/(eol|openuchile|vlabx)-release/(\w+)(.master)?$/, se establece is_release como verdadero. Esto indica que se está realizando un lanzamiento a la rama de producción.

Tengo algunas dudas sobre cómo se maneja esto en los archivos docker.yml de edx-staging y edx-platform.

Por ejemplo, en el archivo docker.yml de edx-platform, encontramos lo siguiente:

`- name: Push the image to the registry
  if: ${{ needs.variables.outputs.is-pr == 'false' }}`

Esto significa que en edx-platform, cuando se llega al paso de pushear la imagen al registro, se evalúa si is_pr es igual a 'false'. Es decir, si no estamos en una PR, en ese caso se ejecutará.

Si is_pr es verdadero, lo que significa que estamos en una PR, entonces este paso no se ejecutará.

Lo mismo sucede con el bloque siguiente en el mismo archivo docker.yml:

`webhook:
  needs: [ variables, build ]
  if: ${{ needs.variables.outputs.is-pr == 'false' }}`

Este bloque tiene una condición que evalúa si is_pr es igual a 'false'. Si es así, se ejecuta este webhook; de lo contrario, se omite. Es decir, se ejecutará si el evento no es una PR.

Este mismo enfoque de condiciones no se encuentra en el archivo docker.yml de edx-staging.

Por ejemplo:

`- name: Login to GitHub Container Registry
  if: github.ref == format('refs/heads/{0}-release/{1}', ${{ env.PROJECT }}, ${{ env.RELEASE-NAME }})

- name: Push
  if: github.ref == format('refs/heads/{0}-release/{1}', ${{ env.PROJECT }}, ${{ env.RELEASE-NAME }})

webhook:
  needs: [ variables, build ]`

Aquí, las tareas de "login to GHCR" y "Push" se ejecutan si la rama actual coincide con el formato especificado dentro del bloque if. Los valores para esta coincidencia se extraen de las variables de entorno PROJECT y RELEASE-NAME, así como CODE y TS.